docs/architecture.md

```diff
@@ -0,0 +1,223 @@
+
+X, Y, and Z are separated, because SIMD computations favor each to be on their own buffer.
+
+CoordinateSpaces are stored in the Universe and are addressable with an index.
+
+Two important Universe components are used to track CoordinateSpace data:
+* **UCompInCoordspace**: Stores the index to which CoordinateSpace a Satellite is part of
+* **UCompCoordspaceIndex**: Stores the Satellite's index inside the CoordnateSpace, used to access CComps.
+
+See [CartesianSimple.h](https://github.com/TheOpenSpaceProgram/osp-magnum/blob/master/src/osp/CoordinateSpaces/CartesianSimple.h) for how CoordinateSpace data can be stored
+
+See [coordinates.h](https://github.com/TheOpenSpaceProgram/osp-magnum/blob/master/src/osp/coordinates.h) for the CoordinateSpaces themselves
+
+### ActiveArea
+
+The ActiveArea is the name given to a small area in the Universe synchronized with the game engine. This is represented as a Satellite with a **UCompActiveArea**. UCompActiveArea stores queues to communicate with the game engine. The game engine can tell the ActiveArea where to move, allowing the user to explore and interact with the universe in-game. This also elegantly solves the floating origin problem.
+
+ActiveArea keeps track of and informs the game engine on Satellites that have entered and exited the area, by determining which ones meet certain conditions to make them visible (such as being nearby). The game engine is responsible for reading the Satellite data, and loading an in-game representation.
+
+For example, if the ActiveArea goes near a planet, it will notify the game engine, which will read the Satellite's data and maybe load terrain and an atmosphere. If a vehicle enters the ActiveArea, then the game engine will read the data and create a physical Vehicle.
+
+ActiveArea also relies on a few of its own CoordinateSpaces to be able to move around freely and also 'capture' other Satellites. This won't be discussed here in detail.
+
+### Universe updates
+
+Universe on its own simply provides data storage. It is the user application's responsibility to pass pieces of Universe into update functions.
+
+Not much is written yet. Coordinate spaces are intended to be passed to update functions to step them through time.
+
+## ActiveScene rundown
+
+ActiveScene stores the state of a 3D game scene, and comes with almost absolutely nothing on its own. All features are 'add-ons' in the form of ECS system functions that need the right components setup in the scene, best explained individually. It does not provide its own window API or anything like that; this is the responsibility of the user's application.
```

Comment:
I recommend discussing what the difference between an ActiveArea and an ActiveScene are